tensorflow/contrib/tensorrt/kernels/trt_calib_op.cc

```diff
@@ -0,0 +1,134 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/contrib/tensorrt/kernels/trt_calib_op.h"
+#include "tensorflow/contrib/tensorrt/resources/TRTInt8Calibrator.h"
+#include "tensorflow/contrib/tensorrt/resources/TRTResourceManager.h"
+#include "tensorflow/contrib/tensorrt/resources/TRTResources.h"
+#include "tensorflow/core/framework/node_def.pb.h"
+#include "tensorflow/core/framework/tensor.h"
+#include "tensorflow/core/framework/tensor_shape.h"
+#include "tensorflow/core/framework/tensor_types.h"
+#include "tensorflow/core/framework/types.h"
+
+#if GOOGLE_CUDA
+#if GOOGLE_TENSORRT
+#include "cuda_runtime_api.h"
+#include "tensorrt/include/NvInfer.h"
+
+namespace tensorflow {
+namespace trt {
+TRTCalibOp::TRTCalibOp(OpKernelConstruction* context) : OpKernel(context) {
+  OP_REQUIRES_OK(context, context->GetAttr("segment_nodes", &segment_nodes_));
+  OP_REQUIRES_OK(context, context->GetAttr("input_names", &input_names_));
+  OP_REQUIRES_OK(context, context->GetAttr("resource_name", &resource_name_));
+};
+
+//  case statement for type
+#define TYPECASE(dt, X, Y)                                             \
+  case dt: {                                                           \
+    Y = (void*)X->flat<tensorflow::EnumToDataType<dt>::Type>().data(); \
+    break;                                                             \
+  }
+
+// macro to get tensor data address pointed by tensor_ptr into
+// void* dest_ptr
+#define GET_TENSOR_ADDRESS(tensor_ptr, dest_ptr)               \
+  {                                                            \
+    auto TENSOR_TYPE = tensor_ptr->dtype();                    \
+    switch (TENSOR_TYPE) {                                     \
+      TYPECASE(tensorflow::DT_FLOAT, tensor_ptr, dest_ptr);    \
+      TYPECASE(tensorflow::DT_HALF, tensor_ptr, dest_ptr);     \
+      TYPECASE(tensorflow::DT_INT8, tensor_ptr, dest_ptr);     \
+      default: {                                               \
+        LOG(FATAL) << "Unsupported Data type "                 \
+                   << tensorflow::DataTypeString(TENSOR_TYPE); \
+        break;                                                 \
+      }                                                        \
+    }                                                          \
+  }
+
+void TRTCalibOp::Compute(tensorflow::OpKernelContext* ctx) {
+  auto trt_rm = tensorflow::trt::TRTResourceManager::instance();
+  auto resmgr = trt_rm->getManager("TRTCalibOps");
+  tensorflow::trt::TRTCalibrationResource* calibRes = nullptr;
+  auto status = resmgr->Lookup(resource_name_, resource_name_, &calibRes);
+
+  if (!status.ok()) {
+    ctx->SetStatus(status);
+    return;
+  }
+  int numInputs = ctx->num_inputs();
```

Comment:
Please follow the style guide to fix the variable names.

Reply:
Should be done